pkg/validations/validations.go

```diff
@@ -0,0 +1,63 @@
+package validations
+
+import (
+	"reflect"
+	"strconv"
+
+	"github.com/openshift-online/ocm-csv-parser/pkg/errormsgs"
+	errors "github.com/zgalor/weberr"
+)
+
+func validateDataType(resourceId, resourceField string, resourceValue interface{}, dataType reflect.Type) error {
+	if reflect.TypeOf(resourceValue) != dataType {
+		return errors.UserErrorf(errormsgs.SyntaxDataType, resourceId, resourceField)
+	}
+	return nil
+}
+
+func validateMachineTypeDataTypes(resource map[string]interface{}) error {
```

Comment:
can we use constant for the field names?

Reply:
Good idea, i'll add that